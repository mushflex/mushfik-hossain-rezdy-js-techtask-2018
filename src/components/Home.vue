<template>
	<div class="home">
		<button class="btn btn-upload grey" @click="goToList">
			<label>{{buttonLabel}}</label>
		</button>
	</div>
</template>

<script>

export default {
	name: 'Home',
	props: [ 'buttonLabel' ],
	mounted: function () {
		// Instantiate
		this.buttonLabel = 'Whats for lunch';
	},
	methods: {
		goToList: function () {
			this.$router.push('/recipes');
		}
	}
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
.header {
	padding: 0.5rem;
	background: #7344c0;
}
h1 {
	color: white;
}
.sub {
	color: white;
}
</style>
