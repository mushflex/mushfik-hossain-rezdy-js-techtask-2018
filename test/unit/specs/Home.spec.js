import Vue from 'vue'
import Home from '@/components/Home'

describe('Home.vue', () => {
	it('has the expected html structure', () => {
		const Constructor = Vue.extend(Home);
		const vm = new Constructor().$mount();
		expect(vm.$el).toMatchSnapshot();
	})

	it('should render button', () => {
		const Constructor = Vue.extend(Home)
		const btnLabel = 'Whats for lunch'
		const vm = new Constructor().$mount()
		console.log(vm.$el.querySelector('.main-button').textContent)
		expect(vm.$el.querySelector('.main-button').textContent)
			.toEqual(btnLabel)
	})
})
