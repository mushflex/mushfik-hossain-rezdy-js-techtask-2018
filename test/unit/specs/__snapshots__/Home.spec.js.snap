// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue has the expected html structure 1`] = `
<div
  class="home"
>
  <button
    class="main-button"
  >
    <label>
      Whats for lunch
    </label>
  </button>
</div>
`;
